### DISCUSSION TITLE
How to rename pipeline ?

### DISCUSSION THREAD

**Discussion by zszen** [gocd/gocd#12703] [Q&A]
How to rename pipeline. Now I can only duplicate with new name and remove old one for it.

**Comment by chadlwilson** [comment]
Unfortunately it's not possible to rename pipelines while retaining the history. The pipelines name is treated as its unique identifier.

If you use pipelines-as-code this is effectively handled for you when you rename, but you'll still lose the run history after a rename.

**Reply by zszen** [reply]
Perhaps it will be a feature that can be achieved in the future

**Reply by chadlwilson** [reply]
It's unlikely - use of the pipeline name as primary key is too baked into the fundamental design of GoCD given its currently maintenance/maintainer state to make such a big change.
